pkg/cpio/cpio.go

```diff
@@ -77,9 +78,28 @@ func (r Reader) ReadRecords() ([]Record, error) {
 
 type Writer struct {
 	rw RecordWriter
+	// There seems to be no harm done in stripping
+	// duplicate names when the record is written,
+	// and lots of harm done if we don't do it.
+	alreadyWritten map[string] bool
```

Comment:
yeah, tbh I've always thought that was one idiom that was ugly as hell, but I'll do it.